Add rules for aep-140 with docs and tests (#35)
aep-dev/aep-openapi-linter: docs/0140.md

# Rules for AEP-140: Field Names

[aep-140]: https://aep.dev/140

> ## aep-140-boolean-field-naming
> 
> **Rule**: Boolean fields should omit the prefix "is".

Reviewer: ```suggestion
## aep-140-boolean-property-naming

**Rule**: Boolean properties should omit the prefix "is".
```

This rule enforces that all boolean properties don't start with the prefix
"is".

### Details

This rule looks for "boolean" properties and complains if the 'boolean' field
name starts with the prefix "is".

### Examples

**Incorrect** code for this rule:

```yaml

> 'schemas':

Reviewer 2: the other examples don't quote the keys (similar to almost every other yaml example I've seen) - I think it'd be good to follow the existing style unless there's a reason not to.

  {
    'book':
      {
        'type': 'object',
        'properties':
          {
            'isbn': { 'type': 'array', 'items': { 'type': 'string' } },
            'is_published': { 'type': 'boolean' }
```

**Correct** code for this rule:

```yaml
'schemas':
  {
    'book':
      {
        'type': 'object',
        'properties':
          {
            'isbn': { 'type': 'array', 'items': { 'type': 'string' } },
            'published': { 'type': 'boolean' }
```

### Disabling

If you need to violate this rule for a specific operation, add an "override" to
the Spectral rule file for the specific file and fragment.

```yaml
overrides:
  - files:
      - 'openapi.json#/components/schemas/book/properties/is_published'
    rules:
      aep-140-boolean-field-naming: 'off'
```

> ## aep-140-uri-field-naming
> 
> **Rule**: Fields representing URLs or URIs should be named "uri" rather than

Reviewer: ```suggestion
## aep-140-uri-property-naming

**Rule**: Properties representing URLs or URIs should be named "uri" rather than
```

"url".

> This rule checks that all "url" fields are named in accordance with the

Reviewer: ```suggestion
This rule checks that all "url" properties are named in accordance with the
```

[AEP-140] ruleset.

### Details

This rule complains if there are any string properties named "url".

### Examples

**Incorrect** code for this rule:

```yaml
'schemas':
  {
    'book':
      {
        'type': 'object',
        'properties':
          {
            'url': { 'type': 'string', 'readOnly': true },
            'published': { 'type': 'boolean' }
```

**Correct** code for this rule:

```yaml
'schemas':
  {
    'book':
      {
        'type': 'object',
        'properties':
          {
            'uri': { 'type': 'string', 'readOnly': true },

>             'published': { 'type': 'boolean'

Reviewer 2: ```suggestion
            'published': { 'type': 'boolean' }
```
